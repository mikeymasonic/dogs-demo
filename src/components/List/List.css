.List {
  
  input[type='radio'] {
    display: none;
  }

  label {
    display: flex;
    color: powderblue;
    font-weight: bolder;
    font-size: 1rem;
    width: 15rem;
    height: 3rem;
    max-width: 15rem;
    margin: .5rem;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    border: dotted .1vw rgb(197, 197, 197);
    background-color: purple;
  }

  label:hover {
    background: #004c97;
    color: greenyellow;
    font-weight: bolder;
    width: 15rem;
    height: 3rem;
    margin: .5rem;
  }
  
  input[type='radio']:checked + label {
    background: #004c97;
    color: greenyellow;
    width: 15rem;
  }
  
}