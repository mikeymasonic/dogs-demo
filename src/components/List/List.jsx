import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const List = ({ breed }) => (
  <section>
    <p><Link to={`detail/${breed}`} >{breed}</Link></p>
  </section>
);

List.propTypes = {
  breed: PropTypes.string,
};

export default List;
