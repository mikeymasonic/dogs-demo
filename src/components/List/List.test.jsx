import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List
      breed="husky"
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
