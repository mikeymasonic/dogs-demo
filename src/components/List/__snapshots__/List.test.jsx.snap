// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<section>
  <p>
    <Link
      to="detail/husky"
    >
      husky
    </Link>
  </p>
</section>
`;
